### PR TITLE
Support code injection in webpack 5

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
The webpack-dash-dynamic-import plugin was written with Webpack 4.x in mind. With Webpack 5.x the API for plugins changed in a way that makes the existing code injection produce invalid JS bundles. This PR makes the injection work with both 4.x and 5.x.

Related to https://github.com/plotly/dash-table/pull/875
Sanity check against Webpack 4.x project: https://app.circleci.com/pipelines/github/plotly/dash-core-components?branch=sanity-webpack-4and5

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Nice find! Can you point to where this change is documented by Webpack? And I presume with this change we will support independently upgrading the renderer and individual component packages from 4 to 5?

File: @plotly/webpack-dash-dynamic-import/LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2019 Plotly
```
Comment (alexcjohnson) [inline @plotly/webpack-dash-dynamic-import/LICENSE:3]: FYI, as I wrote in https://github.com/plotly/dash-core/issues/324

> I've seen some major packages like React drop the copyright year - that simplifies managing these and apparently doesn't have a strong downside, so I'd propose we do the same.

**Comment by Marc-Andre-Rivet** [comment]
> Nice find! Can you point to where this change is documented by Webpack? And I presume with this change we will support independently upgrading the renderer and individual component packages from 4 to 5?

I can't find a reference to this in the plugins documentation but it seems to boil down to a change in the internal implementation with `Template.asString(<array>)` being resolved in 4.x with https://github.com/webpack/webpack/blob/444e59f8a427f94f0064cae6765e5a3c4b78596d/lib/Template.js#L164 and being resolved with `${array}` in 5.x https://github.com/webpack/tapable/blob/master/lib/SyncWaterfallHook.js#L17

File: @plotly/webpack-dash-dynamic-import/LICENSE (hunk 1 above)
Comment (Marc-Andre-Rivet) [reply @plotly/webpack-dash-dynamic-import/LICENSE:3]: Will bump to 2021 for now and until the discussion on this happens

**Comment by alexcjohnson** [comment]
OK so if in 4.x the array was eventually evaluated as `source + '\n' + resolveImportSource()`, and now in 5.x we can't do that but `source + resolveImportSource()` works, is there one of those two flavors that works in both 4 and 5, so we don't need to branch?

**Comment by Marc-Andre-Rivet** [comment]
> OK so if in 4.x the array was eventually evaluated as `source + '\n' + resolveImportSource()`, and now in 5.x we can't do that but `source + resolveImportSource()` works, is there one of those two flavors that works in both 4 and 5, so we don't need to branch?

Yes 😁

File: @plotly/webpack-dash-dynamic-import/src/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const fs = require('fs');
+const webpack = require('webpack');
```
Comment (alexcjohnson) [inline @plotly/webpack-dash-dynamic-import/src/index.js:2]: 🔪

**Review by alexcjohnson — APPROVED** [review]
Beautiful 💃 
Just one little 🔪 comment

**Pull request merged**